### PR TITLE
Adapt registration and submit log to dark theme

### LINKED ISSUE #2625: Minor theme inconsistency
When changing the theme to 'Dark' before registering a phone number, and then going into settings and pressing 'Push messages' (the first option), a window with the heading 'Connect With TextSecure' opens which appears to use the default light theme, which appears to be inconsistent with the rest of the app.

This also appears to be an issue for the 'Change passphrase' window found under Settings / App protection / Enable Passphrase and the 'Submit debug logs' window under advanced settings.

Saumsung SGH-T399
Android 4.2.2
Textsecure 2.6.1

### PR DESCRIPTION
**Pull request by FeuRenard**
### First time contributor checklist
- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Sony Xperia U, Android 4.4.4 (screenshots)
 * AVD Nexus 5, Android 6.0
- [x] My contribution is fully baked and ready to be merged as is
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

----------

### Description
I added `DynamicTheme` to three activities. Then I replaced the `background_pane`s with `CardView`s, because there was an inappropriate shadow in dark theme. In this process I also updated some icons, because they were outdated, inconsistent and did not fit to dark theme.

Roughly 80% of the additions are caused by indentation, most ones in xml. I can adapt it to a guideline, if you hand me some for your xml-files. But the effect of introducing `CardView`s on indentation will be large anyway.

Possible further improvements, that I did not apply:
- RTL support for registration
- white phone icon (32dp)
- round corners of `CardView`s ?
- center alert headline vertically in Android 5 and 6
marshmallow-centering

Fixes #2625

### Screenshots
log-dark
enter_number-dark
connectivity_failure-dark
connectivity_failure-light
verification_failure_dark
verification_failure-light
verifying-dark
verifying-light

### CODE REVIEW

File: src/org/thoughtcrime/securesms/LogSubmitActivity.java [hunk 1]
```diff
@@ -27,6 +31,7 @@ protected void onCreate(Bundle icicle) {
 
   @Override
   protected void onResume() {
+    dynamicTheme.onCreate(this);
```
Comment (cascheberg) [inline src/org/thoughtcrime/securesms/LogSubmitActivity.java]: I haven't actually tested the code... but:
1. should use `onResume` here?
2. in `RegistrationActivity` there is no call to `dynamicTheme.onResume`

**Comment by RiseT** [comment]
Excuse my pretty dumb question, but are these "before" or "after" screenshots? If the latter, **"Possible problems"** is colored incorrectly in both light and dark theme (should be "you know what I mean"-blue).

**Comment by FeuRenard** [comment]
@RiseT All screenshots ar "after" screenshots and show only activities that are affected by this PR. I fixed the link color.

### Screenshots
link-dark
link-light

**Comment by haffenloher** [comment]
The check marks in the last two screenshots of your original post look slightly shifted to the bottom. Don't know if it has always been like that?

**Comment by FeuRenard** [comment]
The progress indicator is also shifted down. But I did not touch the placing of check marks and progress indicators and therefore did not introduce this issues, I think.

I am going to have a look at the alignment of the text.

**Comment by FeuRenard** [comment]
The last commit improves the text alignment in Android 5 and 6, but for Android 4.4 nothing changes. But I tried the current master and I can say that this PR's text alignment is even slightly better although not perfect (4.4).

**Comment by RiseT** [comment]
I don't know what's the "official" way of placing check marks on Android or if this is easy to do in this case, but imho checkmarks shouldn't be vertically centered but aligned to the bottom.

See [this](https://en.wikipedia.org/wiki/Check_mark) Wikipedia page for examples.

Just a thought.

**Comment by FeuRenard** [comment]
Thank you for your input. Personally, I prefer to have the check marks as they are now. And I don't want to lose sight of this PR's scope (adapt to dark theme).

**Comment by moxie0** [comment]
This is a pretty big diff.  If you want to submit a separate PR for the debug log theming, that looks easy to get in.  For the registration screen, I'd also like to split up the PR for updating the assets/look and supporting themes (the latter is extremely low on my list of priorities).

For updating the look, we're generally trying to go with the technique of using white assets and tinting them rather than having separate assets for light and dark.  Focused/unfocused text in the progress activity seems like it could be nice, but the strategy of having a whole framelayout with duplicate focused/unfocused textviews seems potentially excessive.  Thanks!

**Pull request closed without merging**